### PR TITLE
python3Packages.dataprep-ml: init at 0.0.18

### PR DESCRIPTION
**Pull request by mbalatsko**
## Description of changes
Added packages with data utilities for Machine Learning pipelines used by `mindsdb`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mbalatsko** [comment]
Dependency of https://github.com/NixOS/nixpkgs/issues/247089

**Comment by mbalatsko** [comment]
Result of `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` of x86_64-linux:

```
2 packages built:
python310Packages.dataprep-ml python311Packages.dataprep-ml
```

**Comment by mbalatsko** [comment]
@GrahamcOfBorg build python310Packages.dataprep-ml python311Packages.dataprep-ml

File: pkgs/development/python-modules/dataprep-ml/default.nix [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+let
+  testNltkData = symlinkJoin {
+    name = "nltk-test-data";
+    paths = [ nltk-data.punkt nltk-data.stopwords ];
+  };
+in
+buildPythonPackage rec {
+  pname = "dataprep-ml";
+  version = "0.0.18";
+  format = "pyproject";
```
Comment (natsukium) [inline pkgs/development/python-modules/dataprep-ml/default.nix:27]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/dataprep-ml/default.nix [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    "dataprep_ml.cleaners"
+    "dataprep_ml.helpers"
+    "dataprep_ml.imputers"
+    "dataprep_ml.insights"
+    "dataprep_ml.recommenders"
+    "dataprep_ml.splitters"
+  ];
+
+  meta = with lib; {
+    description = "Data utilities for Machine Learning pipelines.";
```
Comment (natsukium) [inline pkgs/development/python-modules/dataprep-ml/default.nix:72]:
```suggestion
    description = "Data utilities for Machine Learning pipelines";
```

File: pkgs/development/python-modules/dataprep-ml/default.nix [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    pydantic
+  ];
+
+  # PyPI tarball has no tests
+  doCheck = false;
+
+  # Package import requires NLTK data to be downloaded
+  # It is the only way to set NLTK_DATA environment variable,
+  # so that it is available in pythonImportsCheck
+  env.NLTK_DATA = testNltkData;
```
Comment (natsukium) [inline pkgs/development/python-modules/dataprep-ml/default.nix:60]: It's a bit tricky.
I feel like we need to do something about type-infer.

File: pkgs/development/python-modules/dataprep-ml/default.nix (hunk 2 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/dataprep-ml/default.nix]: fixed

File: pkgs/development/python-modules/dataprep-ml/default.nix (hunk 1 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/dataprep-ml/default.nix]: Fixed

File: pkgs/development/python-modules/dataprep-ml/default.nix (hunk 3 above)
Comment (mbalatsko) [reply pkgs/development/python-modules/dataprep-ml/default.nix:60]: I haven't come up with the solution unfortunately :( nltk data are by design nix unfriendly. https://github.com/nltk/nltk/blob/e2d368e00ef806121aaa39f6e5f90d9f8243631b/nltk/data.py#L89-L97

Comment (mbalatsko) [reply pkgs/development/python-modules/dataprep-ml/default.nix:60]: It searches for them in these dirs or in the dir specified in NLTK_DATA env variable, so I don't really know how to bake these data inside nix derivation

**Comment by natsukium** [comment]
@ofborg eval

**Review by natsukium — APPROVED** [review]
I'll merge it after ofborg's evaluation.

**Pull request merged**